crates/biome_js_analyze/src/lint/nursery/no_label_without_control.rs

```diff
@@ -0,0 +1,290 @@
+        _ => None,
+    }
+}
+
+/// Returns whether the passed `AnyJsxTag` have a `for` or `htmlFor` attribute
+fn has_for_attribute(jsx_tag: &AnyJsxTag) -> Option<bool> {
+    let for_attributes = &["for", "htmlFor"];
+    let attributes = get_element_attributes(jsx_tag)?;
+
+    Some(attributes.into_iter().any(|attribute| {
+        match attribute {
+            AnyJsxAttribute::JsxAttribute(jsx_attribute) => jsx_attribute
+                .name()
+                .is_ok_and(|jsx_name| for_attributes.contains(&jsx_name.text().as_str())),
+            _ => false,
+        }
+    }))
+}
+
+/// Returns whether the passed `AnyJsxTag` have a child that is considered a control component
+/// according to the passed `control_components` parameter
+fn has_nested_control(jsx_tag: &AnyJsxTag, control_components: &[String]) -> bool {
+    jsx_tag.syntax().descendants().any(|descendant| {
+        match (
+            JsxName::cast(descendant.clone()),
+            JsxReferenceIdentifier::cast(descendant.clone()),
+        ) {
+            (Some(jsx_name), _) => control_components.contains(&jsx_name.text()),
+            (_, Some(jsx_reference_identifier)) => {
+                control_components.contains(&jsx_reference_identifier.text())
+            }
+            _ => false,
+        }
+    })
+}
+
+/// Returns whether the passed `AnyJsxTag` meets one of the following conditions:
+/// - Has a label attribute that corresponds to the `label_attributes` parameter
+/// - Has an `aria-labelledby` attribute
+/// - Has a child `JsxText` node
+fn has_accessible_label(jsx_tag: &AnyJsxTag, label_attributes: &[String]) -> bool {
+    let jsx_attributes = jsx_tag
+        .syntax()
+        .descendants()
+        .filter_map(JsxAttribute::cast)
+        .collect::<Vec<JsxAttribute>>();
+
+    let has_accessible_attribute = jsx_attributes.iter().any(|jsx_attribute| {
+        match (
+            jsx_attribute.name().ok(),
+            jsx_attribute
+                .initializer()
+                .and_then(|initializer| initializer.value().ok()),
+        ) {
+            (Some(AnyJsxAttributeName::JsxName(jsx_name)), Some(jsx_attribute_value)) => {
+                let has_label_attribute = label_attributes.contains(&jsx_name.text());
+                let is_aria_labelledby_attribute = jsx_name.text() == "aria-labelledby";
+                let has_value = has_jsx_attribute_value(jsx_attribute_value);
+
+                has_value && (is_aria_labelledby_attribute || has_label_attribute)
+            }
+            _ => false,
+        }
+    });
+
+    let has_text = jsx_tag
+        .syntax()
+        .descendants()
+        .any(|descendant| JsxText::cast(descendant).is_some());
+
+    has_accessible_attribute || has_text
+}
+
+/// Returns whether the passed `jsx_attribute_value` has a valid value inside it
+fn has_jsx_attribute_value(jsx_attribute_value: AnyJsxAttributeValue) -> bool {
```

Comment:
We have an utility called `as_static_value` (`StaticValue`). I think that's what you're looking for. Also, `to_string` allocates a new string, and we don't need it

Reply:
forgot about this one, done!